### DISCUSSION TITLE
How can I retrieve the response body data as a parameter?

### DISCUSSION THREAD

**Discussion by heyyuyu** [mockoon/mockoon#991] [Help]
How can I retrieve the response body data as a parameter?

**Comment by 255kb** [comment]
I am not sure I understand what you are trying to achieve exactly, without more details. Could you please provide the setup you have in Mockoon and an example of a request you make to the mock?

**Reply by heyyuyu** [reply]
for example,I try to mock response like this:
`{
    "code": 0,
    "data":[
     {{# repeat 0 (queryParam 'pageSize' 20)}}
         {
              "name":"XXX",
              "age": 12
         }
     {{/ repeat}}
    ],
    "total": 5
}`
and I want to pass the size of “data“ as a parameter to the variable "total".can I retrieve that？thanks！

**Reply by 255kb** [reply]
Yes you can, by calling the endpoint with a query parameter: 
`/endpoint?pageSize=50`

**Reply by heyyuyu** [reply]
> Yes you can, by calling the endpoint with a query parameter: `/endpoint?pageSize=50`

It does not work,since the “data" is random, it is not possible to determine the size of the data using pageSize.

**Reply by 255kb** [reply]
sorry I misunderstood your question. 
It's true, with `#each` you have access to `index` but not with `repeat`. 
What you can do is to create a variable using [`setVar`](https://mockoon.com/docs/latest/templating/mockoon-helpers/#setvar):
```
{{ setVar 'total' (queryParam 'pageSize' 20) }}
{
  "code": 0,
  "data":[ {{# repeat 0 @total}} { "name":"XXX", "age": 12 } {{/ repeat}} ], 
  "total": @total 
}
```
